lldb/tools/lldb-dap/DAP.cpp

```diff
@@ -893,10 +893,23 @@ llvm::Error DAP::Loop() {
             return errWrapper;
           }
 
+          // The launch sequence is special and we need to carefully handle
+          // packets in the right order. The launch and attach requests cannot
+          // be answered until we've gotten the confgigurationDone request. We
+          // can't answer the threads request until we've successfully launched
+          // or attached.
+          bool is_part_of_launch_sequence = false;
```

Comment:
After `request initialize` + `event initialized`, should we queue everything until we see `configurationDone`? We're already making a special case for `threads` and I worry about other special cases. Maybe the general flow should be to put all messages into the launch queue except `initialize`, `configurationDone` and `disconnect` (in case we need to shutdown before launching is complete) until we've responded to `configurationDone`. Then we can process the launch queue and processed as normal at that point.

Reply:
We don't have to respond to those requests immediately, they're sent, but at least VSCode doesn't wait for a response to them before sending the `configurationDone` request.

Reply:
I guess that would be a problem though because we'd miss hitting initial breakpoints, so ya, your right about those requests.